### PR TITLE
Add some logging about the runtime config + feature gates when the container is loaded

### PR DESCRIPTION
**Pull request by andre4i**
## Description

As more of our scenarios are rolling out using custom configs and possibly feature gates, it would be very useful for an oncall engineer to consult the log and get a full view (as much as already instrumented, new features gates must be explicitly added to the log) of what is enabled/disabled and what is the final config the runtime is running with.

### CODE REVIEW

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 1]
```diff
@@ -1416,6 +1427,13 @@ export class ContainerRuntime
 			summaryFormatVersion: metadata?.summaryFormatVersion,
 			disableIsolatedChannels: metadata?.disableIsolatedChannels,
 			gcVersion: metadata?.gcFeature,
+			options: JSON.stringify(runtimeOptions),
+			featureGates: JSON.stringify({
```
Comment (vladsud) [inline packages/runtime/container-runtime/src/containerRuntime.ts:1431]: I wonder, is there a way to enumerate all the gates out there, even if code does not know about their names? Seems like better direction (maybe we attack it as separate task)

**Review by vladsud — APPROVED** [review]

File: packages/runtime/container-runtime/src/containerRuntime.ts (hunk 1 above)
Comment (andre4i) [reply packages/runtime/container-runtime/src/containerRuntime.ts:1431]: Yes! I mean there should be, and it has been an ask about this, which I explored briefly but ran into some chicken-and-egg problems about logging stuff within the logger (mixin config logger, actually). The gist of it is that we can capture all feature gate 'get' calls and log the first time they're accessed.

Unfortunately, there is no central place for feature gates to be declared, but there probably should be, something similar to what we do with the assert codes.

Comment (andre4i) [reply packages/runtime/container-runtime/src/containerRuntime.ts:1431]: Opened ADO:3466

Comment (vladsud) [reply packages/runtime/container-runtime/src/containerRuntime.ts:1431]: We should double check with Loop (the bottom of the stack). I hope there is a way to enumerate all feature gates, as they are known (i.e. we get whole package from ECS, it's not call-per-call retrieval).
Logging get calls will not work - some of the get calls may happen minutes into the session.

**Pull request merged**